src/useLocalStore.ts

```diff
@@ -11,9 +11,13 @@ function wrapInTransaction(fn: Function) {
     }
 }
 
-export function useLocalStore<T>(initializer: () => T): T {
-    return useState(() => {
-        const store: any = observable(initializer())
+export function useLocalStore<T>(initializer: (props?: any) => T, current?: any): T {
```

Comment:
I think this type suffices:
`export function useLocalStore<T, P = any>(initializer: (props?: P) => T, current?: P): T {`